crates/bevy_app/src/plugin.rs

```diff
@@ -1,8 +1,16 @@
+#[cfg(feature = "downcast")]
 use downcast_rs::{impl_downcast, Downcast};
 
 use crate::App;
 use core::any::Any;
 
+#[cfg(not(feature = "downcast"))]
+#[doc(hidden)]
+pub trait Downcast {}
+
+#[cfg(not(feature = "downcast"))]
+impl<T: ?Sized> Downcast for T {}
```

Comment:
`downcast-rs` isn't compatible with `portablie-atomic` (though it could be with a PR I believe!), so I've included a little dummy trait so the `Plugin: Downcast` bound can remain.